tools/python/timestream-compute-units-testing/last_point-query.ipynb

```diff
@@ -0,0 +1,187 @@
+{
+ "cells": [
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e33a646c",
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "import boto3\n",
+    "import threading\n",
+    "import time\n",
+    "from datetime import datetime\n",
+    "import numpy as np\n",
+    "import matplotlib.pyplot as plt\n",
+    "from botocore.exceptions import ClientError\n",
+    "\n",
+    "# Initialize the Timestream client\n",
+    "client = boto3.client('timestream-query', region_name='us-east-2')\n",
+    "\n",
+    "# List of different Timestream queries\n",
+    "queries = [\n",
+    "    f'select memory from \"devops\".\"sample_devops\" where time > ago(10m) and hostname=\\'host1\\' order by time desc limit 1',\n",
+    "]\n",
+    "\n",
+    "# Function to run a single query and measure its duration\n",
+    "def run_query(query, worker_id, worker_results,throttling_results):\n",
+    "    start_time = time.time()\n",
+    "    worker_results[worker_id] = []\n",
+    "    end_time = time.time() + duration_seconds\n",
+    "    throttling_results[worker_id] = []\n",
```

Comment:
Nit. But for clarity. you can refactor this code. something like this..
def run_query(query, worker_id, worker_results, throttling_results):
    start_time = time.time()
    end_time = start_time + duration_seconds
    worker_results[worker_id] = []
    throttling_results[worker_id] = []

    while time.time() < end_time:
        try:
            response = execute_query(query)
            worker_results[worker_id].append(response.duration)
        except ClientError as e:
            handle_client_error(e, worker_id, throttling_results)

    return worker_results, throttling_results

def execute_query(query):
    start_time = time.time()
    response = client.query(QueryString=query)
    return QueryResult(duration=time.time() - start_time)

def handle_client_error(e, worker_id, throttling_results):
    if e.response['Error']['Code'] == 'ThrottlingException':
        throttling_results[worker_id].append("ThrottlingException")
    else:
        raise e